custom_metrics/almanac.js

```diff
@@ -306,6 +359,240 @@ return JSON.stringify({
       return null;
     }
   })(),
-  //  check if there is any picture tag containing an img tag
-  'has_picture_img': document.querySelectorAll('picture img').length > 0
+
+  // Previously for 04_04.sql in 2019
+  'inline_svg_stats': (() => {
+    const svg_elements = [...document.querySelectorAll('svg')];
+
+    return {
+      total: svg_elements.length,
+      content_lengths: svg_elements.map(svg => svg.outerHTML.length),
+      props: parseNodes(svg_elements),
+    };
+  })(),
+
+  // Various stats of img, source and picture elements
+  'images': (() => {
+    const pictures = document.querySelectorAll('picture');
+    const imgs = [...document.querySelectorAll('img')];
+    const sources = document.querySelectorAll('source');
+    const pictures_with_img = document.querySelectorAll('picture img');
+
+    const images_with_srcset = document.querySelectorAll('img[srcset], source[srcset]');
+    const images_with_sizes = [...document.querySelectorAll('img[sizes], source[sizes]')];
+    const images_using_loading_prop = [...document.querySelectorAll('img[loading], source[loading]')];
+
+    // NOTE: -1 is used to represent images with no alt tag at all. Empty alt tags have a value of 0
```

Comment:
I specifically identify missing and blank as separate things, as missing is an error, blank is valid. length stats may be of interest, but has not been asked for.

images: {
alt: {missing: 3, blank: 1, present: 8},
dimensions: {missing_width: 9, missing_height: 10},
loading: {auto: 2, lazy: 1, eager: 1, invalid: 1, missing: 6, …},
total: 12
}

Reply:
The amount of data returned from images has been reduced to only specific properties. As for aggregating the stats, thats quite difficult to do. You cant progressively compile a median that way, since you need the whole dataset

Reply:
I'm assuming our analysis will be more about identifying pages with non zero scores. e.g. pages using lazy load. Pages with missing alt attributes. If yours returns alt, src, srcset, width, height and loading then we could probably calculate what we need.